### PR TITLE
Update Configstore/index.d.ts with accurate definitions for 2.1.0

### PR DESCRIPTION
**Pull request by ArcticLight**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/yeoman/configstore/blob/master/index.js
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

File: configstore/index.d.ts [hunk 1]
```diff
@@ -1,37 +1,65 @@
-// Type definitions for configstore 0.3.0
+// Type definitions for Configstore 2.1.0
 // Project: https://github.com/yeoman/configstore
-// Definitions by: Bart van der Schoor <https://github.com/Bartvds>
```
Comment (andy-ms) [inline configstore/index.d.ts]: Please add your name to the end: `// Definitions by: Bart van der Schoor <https://github.com/Bartvds>, ArcticLight <https://github.com/ArcticLight/>`, and keep the 4th line of the header.

File: configstore/index.d.ts [hunk 2]
```diff
@@ -1,37 +1,65 @@
...
+    /**
+     * Clear the config.
+     * Equivalent to <code>Configstore.all = {};</code>
+     */
+    clear(): void;
+  }
+
+declare namespace Configstore {
+    export interface ConfigstoreOptions {
+        globalConfigPath?: string;
```
Comment (andy-ms) [inline configstore/index.d.ts]: [docs](https://github.com/yeoman/configstore#globalconfigpath) say it's a boolean.

File: configstore/index.d.ts [hunk 3]
```diff
@@ -1,37 +1,65 @@
...
+    path: string;
+
+    /**
+     * Get all items as an object or replace the current config with an object.
+     */
+    all: any;
+
+    /**
+     * Get an item
+     * @param {string} key The string key to get
```
Comment (andy-ms) [inline configstore/index.d.ts]: JSDoc types are redundant -- that's what TypeScript is for!

**Review by ghost — COMMENTED** [review]
Since you're rewriting the whole file, could you add a `tslint.json` containing `{ "extends": "../tslint.json" }` too?

File: configstore/index.d.ts (hunk 2 above)
Comment (ArcticLight) [inline configstore/index.d.ts]: Fixing it

File: configstore/index.d.ts (hunk 3 above)
Comment (ArcticLight) [inline configstore/index.d.ts]: Fair

**Comment by ArcticLight** [comment]
@andy-ms I finished checking the file against TSLint (and added the new file), can I have you review again? Thank you.

**Pull request merged**